src/streaming/s3.connection.cpp

```diff
@@ -57,7 +57,12 @@ zarr::S3Connection::~S3Connection() = default;
 bool
 zarr::S3Connection::is_connection_valid()
 {
-    return static_cast<bool>(impl_->client->ListBuckets());
+    auto response = impl_->client->ListBuckets();
+    if (response.Error()) {
```

Comment:
I like the logging but the `bool` [cast](https://github.com/minio/minio-cpp/blob/49e578d286cc0abdfbe92cdd5a643ad893b7a619/include/miniocpp/response.h#L61) for the `Response` object is more comprehensive, since it also checks the HTTP status code and not just the error message. 

```suggestion
    const auto response = impl_->client->ListBuckets();
    if (!static_cast<bool>(response)) { // check status code and error message
```